ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -486,10 +491,34 @@ public void destroyConnection(ConnectedChannel connectedChannel) {
     }
   }
 
+  /**
+   * For the given partitionId, construct messages and add them to the given lists.
+   * @param partitionId the {@link PartitionId} to use for generating the {@link BlobId} of the message.
+   * @param messageInfoListofLists the list of {@link List<MessageInfo>} all of which will be populated with the
+   *                               {@link MessageInfo} of the messages.
+   * @param messageBufferListofLists the list of {@link List<ByteBuffer>} all of which will be populated with the
+   *                                 message content of the messages.
+   * @param count the number of messages to construct and add.
+   */
+  private void addMessagesToReplicasOfPartition(PartitionId partitionId,
+      List<ArrayList<MessageInfo>> messageInfoListofLists, List<ArrayList<ByteBuffer>> messageBufferListofLists,
+      int count) throws Exception {
+    for (int i = 0; i < count; i++) {
+      BlobId id = new BlobId(partitionId);
+      ByteBuffer byteBuffer = constructTestBlobInMessageFormat(id, 1000, random);
+      long streamSize = byteBuffer.limit();
```

Comment:
Existing code, and I think `limit()` is okay here depending on how you look at it. Technically the size of the stream is different from what is remaining. For a wrapped `ByteBuffer`, the limit is exactly the size of the data.